src/planning/simple-planner.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Arc} from '../runtime/arc.js';
+import {Recipe} from '../runtime/recipe/recipe.js';
+import {RecipeResolver} from '../runtime/recipe/recipe-resolver.js';
+
+// One entry in the lookup table, i.e. one trigger and the recipe it invokes.
+// Exported only for testing.
+export class Match {
+  constructor(readonly trigger: [string, string][], readonly recipe: Recipe) {}
+
+  // The definition of a matching trigger is embodied in this function.
+  // This first implementation matches if every key-value pair in the trigger
+  // is present in the request. The request may contain other pairs, and the
+  // order of the matching pairs need not be the same as the order in the trigger. 
+  matches(request: [string, string][]): boolean {
+    if (this.trigger.length > request.length) return false;
+    for (const pair of this.trigger) {
+      const found = request.findIndex(r => r[0] === pair[0] && r[1] === pair[1]);
+      if (found === -1) return false;
+    }
+    return true;
+  }
+}
+
+/**
+ * A very simple planner that looks up recipes based on triggers in the
+ * manifest file matching requests.
+ */
+export class SimplePlanner {
+  private _table: Match[] = [];
```

Comment:
nit - `_table` is not a very descriptive name. `matches` maybe? or just `recipeTriggers`? also (nit-nit) you could make it a `public final` and then no getter would be needed.

Reply:
Renamed it to `_recipesByTrigger`. `public final` didn't work, causing the TS compiler to complain. It didn't seem worthwhile to chase it down.